### PR TITLE
Make the block struct repr C and align to 2 bytes

### PR DESCRIPTION
**Pull request by marysaka**
Partially reverts sunriseos/storage_device#4

### CODE REVIEW

File: src/block.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ pub type BlockResult<T> = core::result::Result<T, BlockError>;
 
 /// Represent a certain amount of data from a block device.
 #[derive(Clone)]
-#[repr(C, align(512))]
+#[repr(C)]
```
Comment (Orycterope) [inline src/block.rs]:
```suggestion
#[repr(C, align(2)] // PRDT entries in AHCI must be word aligned
```

Comment (roblabla) [reply src/block.rs]: Is `storage_device` supposed to be tied to AHCI though? I thought it was meant to be an abstraction layer. IMO alignment should be handled on a higher layer.

File: src/block.rs [hunk 2]
```diff
@@ -16,10 +16,10 @@ pub type BlockResult<T> = core::result::Result<T, BlockError>;
...
 pub struct Block {
     /// The actual storage of the block.
-    pub contents: [u8; Block::LEN],
+    pub contents: [u16; Block::LEN / 2],
```
Comment (roblabla) [inline src/block.rs]: Lol... I meant do an u16 array in the .id (which I forgot wasn't possible). This should definitely be `repr(C, align(2))` if we're going to force alignment to 2 here.

**Review by roblabla — CHANGES_REQUESTED** [review]

**Review by roblabla — APPROVED** [review]

File: src/block.rs (hunk 2 above)
Comment (Orycterope) [reply src/block.rs]: 957P24t

**Review by Orycterope — APPROVED** [review]

**Pull request merged**